lib/basic_auth.ex

```diff
@@ -1,11 +1,15 @@
 defmodule BasicAuth do
+  def init(options = {:application_config, _}) do
+    options
+  end
+
   def init(options) do
-    realm    = Keyword.fetch!(options, :realm)
-    username = Keyword.fetch!(options, :username)
-    password = Keyword.fetch!(options, :password)
-    %{realm: realm, username: username, password: password}
+    #Better to fail at compile time if keys are incorrect
+    [:password, :realm, :username] = Keyword.keys(options) |> Enum.sort
```

Comment:
Not really an idiom. I was trying to keep the existing functionality, in which this would fail to compile (as opposed to runtime) if any of the keys were missing.